modules/components/org.wso2.carbon.appfactory.core/src/main/java/org/wso2/carbon/appfactory/core/dao/JDBCResourceDAO.java

```diff
@@ -376,4 +381,237 @@ public boolean updateResource(String applicationKey, String resourceType, String
+
+            //Adding container details
+            while (containerResultSet.next()) {
+                Container container = new Container();
+                container.setBaseImageName(
+                        containerResultSet.getString(SQLParameterConstants.COLUMN_NAME_BASEIMAGE_NAME));
+                container.setBaseImageVersion(
+                        containerResultSet.getString(SQLParameterConstants.COLUMN_NAME_BASEIMAGE_VERSION));
+
+                int containerId = containerResultSet.getInt(SQLParameterConstants.COLUMN_NAME_CONTAINER_ID);
+                List<ServiceProxy> serviceProxies = getServiceProxies(containerId, databaseConnection);
+                container.setServiceProxies(serviceProxies);
+                containers.add(container);
+            }
+
+        } catch (SQLException e) {
+            String message = "Error while getting containers";
+            log.error(message, e);
+        } finally {
+            AppFactoryDBUtil.closeResultSet(containerResultSet);
+            AppFactoryDBUtil.closePreparedStatement(getContainerPreparedStatement);
+        }
+
+        return containers;
+
+    }
+
+    /**
+     * Get service proxies
+     *
+     * @param containerId        id of the container
+     * @param databaseConnection database connetion
+     * @return list of service proxies
+     */
+    private List<ServiceProxy> getServiceProxies(int containerId, Connection databaseConnection) {
+        PreparedStatement getServicePreparedStatement = null;
+        ResultSet serviceResultSet = null;
+        List<ServiceProxy> serviceProxies = new ArrayList<ServiceProxy>();
+
+        try {
+            getServicePreparedStatement = databaseConnection.prepareStatement(SQLConstants.GET_SERVICE_PROXY);
+            getServicePreparedStatement.setInt(1, containerId);
+            serviceResultSet = getServicePreparedStatement.executeQuery();
+
+            //Adding service proxy details
+            while (serviceResultSet.next()) {
+                ServiceProxy serviceProxy = new ServiceProxy();
+                serviceProxy.setServiceName(serviceResultSet.getString(SQLParameterConstants.COLUMN_NAME_SERVICE_NAME));
+                serviceProxy.setServiceProtocol(
+                        serviceResultSet.getString(SQLParameterConstants.COLUMN_NAME_SERVICE_PROTOCOL));
+                serviceProxy.setServicePort(serviceResultSet.getInt(SQLParameterConstants.COLUMN_NAME_SERVICE_PORT));
+                serviceProxy.setServiceBackendPort(
+                        serviceResultSet.getInt(SQLParameterConstants.COLUMN_NAME_SERVICE_BACKEND_PORT));
+                serviceProxies.add(serviceProxy);
+            }
+
+        } catch (SQLException e) {
+            String message = "Error while getting service proxies";
+            log.error(message, e);
+        } finally {
+            AppFactoryDBUtil.closeResultSet(serviceResultSet);
+            AppFactoryDBUtil.closePreparedStatement(getServicePreparedStatement);
```

Comment:
I am closing the db connection after db commit.
https://github.com/wso2/product-af/pull/796/files#diff-b816ace5b656f21ef04cda5e862b059aR466